tools/utility/roles.py

```diff
@@ -0,0 +1,21 @@
+from glob import glob
+import os
+
+import b
+from utility.dockerise import run_on_docker
+
+def dir_role_names(dir):
+    # Find all role files
+    fnames = glob(os.path.join(b.project_dir, "roles", dir, "**", "*.role"), recursive=True)
+    # Strip everything from file paths except role name and subdirectory in roles/
+    return [os.path.splitext(f.replace(os.path.join(b.project_dir, "roles", ""), ""))[0] for f in fnames]
+
+def all_role_names():
+    return dir_role_names("")
+
+def role_folders():
+    # Find all role folders
+    dnames = glob(os.path.join(b.project_dir, "roles", "**/"), recursive=True)
+    dnames.pop(0)
+    # Strip everything from file paths except role name and subdirectory in roles/
+    return [os.path.splitext(d.replace(os.path.join(b.project_dir, "roles", ""), ""))[0] for d in dnames]
```

Comment:
in this part you'll need to remove the trailing `/`

Reply:
I figured those were intentional to differentiate directories from custom groups like the cool_roles and nice_roles but I've removed them now